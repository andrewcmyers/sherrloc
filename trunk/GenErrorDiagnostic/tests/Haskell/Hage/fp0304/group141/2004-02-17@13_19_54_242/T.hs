module T where

splits :: [a] -> ([a],[a])
splits xs = map (\n -> splitAt n xs) [0 length xs]

-- [0 length xs] should be [0 ... length xs]
-- 4,39-49
