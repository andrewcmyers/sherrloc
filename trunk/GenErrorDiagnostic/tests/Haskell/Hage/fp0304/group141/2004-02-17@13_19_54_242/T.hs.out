Compiling T.hs
(4,13)-(4,51): Type error in right-hand side
 expression       : map (\n -> splitAt n xs) [0 length xs]
   type           : [([a], [a])]
   does not match : ([a], [a])  

(4,39)-(4,40): Type error in application
 expression       : 0 length xs
 term             : 0
   type           : Int                       
   does not match : ([a] -> Int) -> [b] -> Int
 because          : it is not a function

Compilation failed with 2 errors
